rotkehlchen/db/calendar.py

```diff
@@ -0,0 +1,217 @@
+from typing import TYPE_CHECKING, Any, NamedTuple
+
+from pysqlcipher3 import dbapi2 as sqlcipher
+
+from rotkehlchen.db.filtering import (
+    DBFilter,
+    DBFilterQuery,
+    DBMultiIntegerFilter,
+    DBOptionalChainAddressesFilter,
+    DBSubStringFilter,
+    DBTimestampFilter,
+    FilterWithTimestamp,
+)
+from rotkehlchen.errors.misc import InputError
+from rotkehlchen.types import (
+    BlockchainAddress,
+    OptionalBlockchainAddress,
+    SupportedBlockchain,
+    Timestamp,
+)
+
+if TYPE_CHECKING:
+    from rotkehlchen.db.dbhandler import DBHandler
+
+
+class CalendarEntry(NamedTuple):
+    """Represents a calendar entry in the database"""
+    name: str
+    timestamp: Timestamp
+    description: str | None
+    counterparty: str | None
+    address: BlockchainAddress | None
+    blockchain: SupportedBlockchain | None
+    identifier: int = 0
+
+    def serialize(self) -> dict[str, Any]:
+        return {
+            'identifier': self.identifier,
+            'name': self.name,
+            'description': self.description,
+            'counterparty': self.counterparty,
+            'timestamp': self.timestamp,
+            'address': self.address,
+            'blockchain': self.blockchain.serialize() if self.blockchain is not None else None,
+        }
+
+    def serialize_for_db(self) -> tuple[
+        str,  # name
```

Comment:
why not put identifier here first too and then have it declared as a type and have it the same in serialize and deserialize?